### PR TITLE
Update Cloud API interfaces to use new Region API endpoints

### PR DESCRIPTION
**Pull request by rjobanp**
The new Region API runs in each cloud region and replaces the APIs currently embedded in the Region Controller and Environment Controllers in each cloud region.

This PR updates the `src/cloud-api` crate to point to these new Region API endpoints.

NOTE that this does change the naming of many of the Cloud API client's public methods by re-aligning the naming to use the customer-facing conventions (replace 'environment' with 'region')

Since there are no current uses of the `cloud-api` crate in this repo today, I have not updated any existing tests or references, however this will require a slight re-work of the new MZ CLI PR https://github.com/MaterializeInc/materialize/pull/19340 to switch to the new client methods defined here.

I also skipped updating the endpoints used by the existing CLI code in `src/mz` since that crate is almost entirely re-written in #19340 

### Motivation

  * This PR adds a known-desirable feature: https://github.com/MaterializeInc/cloud/issues/5775

### Tips for reviewer

This is in draft mode and not yet testable until the following Cloud PRs are merged & deployed to our staging environment:
- https://github.com/MaterializeInc/cloud/pull/6961
- https://github.com/MaterializeInc/cloud/pull/6981

Once those are deployed, it may make sense to test this PR by combining with #19340 and leveraging the tests defined there against our staging environment, however I'm open to other ideas on methods to validate this

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: src/cloud-api/src/client/region.rs [hunk 1]
```diff
@@ -8,49 +8,181 @@
...
-    pub environment_controller_url: Url,
+    /// E.g.: 3es24sg5rghjku7josdcs5jd7.eu-west-1.aws.materialize.cloud:443
+    pub http_address: String,
+    /// Indicates true if the address is resolvable by DNS.
+    pub resolvable: bool,
+    /// The time at which the region was enabled
+    pub enabled_at: Option<DateTime<Utc>>,
 }
 
 impl Client {
```
Comment (rjobanp) [inline src/cloud-api/src/client/region.rs:55]: these methods are mostly copies of the removed ones in `environment.rs`, but the diff doesn't show that too well

**Comment by joacoc** [comment]
Super! I'll give a proper review in a bit. In general terms it looks fine

**Review by joacoc — APPROVED** [review]
Looks good to me! :)

**Comment by joacoc** [comment]
@rjobanp This part should be replaced by `api.`, right or do we still have any blocker? https://github.com/MaterializeInc/materialize/blob/514b16322920fb72d0513fbff31a717ab6927f56/src/cloud-api/src/client.rs#L64

**Comment by rjobanp** [comment]
Ah yes good point -- @necaris can you confirm that I should update the URL for the global api? I think your PR https://github.com/MaterializeInc/cloud/pull/6660 made the change but just want to be sure we're confident in doing so now

**Comment by necaris** [comment]
> Ah yes good point -- @necaris can you confirm that I should update the URL for the global api? I think your PR [MaterializeInc/cloud#6660](https://github.com/MaterializeInc/cloud/pull/6660) made the change but just want to be sure we're confident in doing so now

Yep, this should be safe now!

**Review by necaris — APPROVED** [review]

**Review by necaris — APPROVED** [review]

**Comment by rjobanp** [comment]
@joacoc should I merge this as-is? Or do you recommend we wait until https://github.com/MaterializeInc/materialize/pull/19340 has been refactored to work relative to this? Happy to do whatever is easiest for you

**Comment by joacoc** [comment]
@rjobanp Yes, it is ok to merge it as is. There will be no impact on any workflow or CI/CD process. I will pull, merge and test in https://github.com/MaterializeInc/materialize/pull/19340 before replacing the old but current `src/mz`!

**Pull request merged**